### DISCUSSION TITLE
Does vcpkg provide tools or commands to automate or simplify the process of updating your ports in an overlay registry?

### DISCUSSION THREAD

**Discussion by dennisferron** [microsoft/vcpkg#39566] [Q&A]
I created an overlay port (Edit: just realized I typed port here when I meant "overlay registry" or "custom registry" - I'm still unsure the terminology) (forked from North Wind Traders):

https://github.com/dennisferron/vcpkg-registry

I found the process for updating your ports when you update your upstream libraries to be difficult to understand and error prone.  The best documentation I could find about it is here:

https://decovar.dev/blog/2022/10/30/cpp-dependencies-with-vcpkg/

Based on that source, and synthesizing bits and pieces such as Reddit r/cpp comments, I came up with this explanation of the process:

https://github.com/dennisferron/antlr4-testrig-runner/blob/main/HOW_TO_UPDATE_VCPKG_PORT.md

It's still so long, so many manual steps, that I can't figure out a way to shorten this further to fit in a space appropriate for this question.  (Hence why I provide links instead.)

My first question is, do I have this right - what I've written in my how-to at the link, above?

Secondly, are there tools or vcpkg commands I should be aware of that automate some of the steps of this process?

I'm actually working on a tool to automate some or all of it, but I wanted to make sure I'm not duplicating functionality that exists.  And, obviously, that I have the steps right in the first place.

**Comment by dg0yt** [comment]
Please stop missing overlay ports and (git) registries.
(The versioning exists only in git registries.)

Do not use the term "vcpkg repo" when referring to a custom git registry.

The steps in the tutorial aren't in a suitable order. As an abstract guideline:
1. Release your personal project.
2. Update files in `ports/<project>` (incl. `git commit`).
3. update the files in `versions` (git registries only).

The first step is not related to vcpkg. (Automate as you like.)  
The second step is trivial if it is just updating one or two fields in the manifest and one SHA512 in the portfile.  
The last step is a little bit inconvenient, in particular with determining the `git-tree` value. (For the main registry this is handled with `vcpkg x-add-version`.)

**Reply by Osyotr** [reply]
The last step can be simplified by using some vcpkg flags. This is my workflow:
```
cd my-registry
vcpkg format-manifest ports/mypackage/vcpkg.json
git add ports
git commit -m "[mypackage] Update to XYZ"
vcpkg x-add-version --overlay-ports=./ports --x-builtin-registry-versions-dir=./versions/ --x-builtin-ports-root=./ports mypackage
git add versions
git commit --amend --no-edit
```

**Reply by dennisferron** [reply]
> Please stop missing overlay ports and (git) registries. (The versioning exists only in git registries.)
> 
> Do not use the term "vcpkg repo" when referring to a custom git registry.

Thanks; I'm still fairly new to this and struggling to get the terminology down.  Is there a published glossary I could refer to?  If my follow up questions below appear overly pedantic, just know they come from a place of wanting to use the terms correctly.

I assume the injunction to "stop mi[xing up] overlay ports and git registries" refers to how I worded the title?  How would you phrase it better?

The parenthetical that "versioning exists only in Git registries," I would take as either a _non sequitur_ or tautological, but I suspect you meant it as neither.  It might clarify to know if by versioning you mean triplets of three integers called "version" or the source control versioning Git uses hashes to reference.

In re: your second line, I want to accept your point, but I'm not sure where to apply the correction.  The phrase, "vcpkg repo" doesn't actually appear in anything I've written here or at the two Github links in my post.  "Vcpkg repository" does appear in the blog post (not mine) which I linked, but I reviewed it, and (I think) he's using it only to refer to "Microsoft's vcpkg repository" which I assume would be correct usage by you.

Or, is your main objection actually that I _should not have named_ my fork vcpkg-registry in dennisferron/vcpkg-registry?  If so, I apologize; I was merely following a tutorial and, when I forked the North Wind Traders' example, that's what it was named by default.  If that is likely to cause confusion perhaps I should change its name on Github.

In general:  are both "overlay ports" and "overlay registries" terms of art, or only "overlay port"?  If overlay registry isn't a term, what do I properly call the Github repository in which I keep my overlay ports?  (I know Git and Github are not synonyms, but I'm emphasizing that I'm asking what then should I call the artifact at URL https://github.com/dennisferron/vcpkg-registry.)

Finally, what are terms I should use when talking in the same document both about "_my port of_ antlr4-testrig-runner" as it exists within the ports folder of my custom registry and also the upstream project the port pulls in?  And at what point does the word "package" become appropriate, if ever?
